### PR TITLE
Refactoring: Remove all references to prodtype (code/tests/docs)

### PR DESCRIPTION
**Pull request by evgenyz**
#### Description:

Pretty much what it says on the tin. After merging #11378 we don't have any use for `prodtype` as it always equals *all*.

This PR removes (and occasionally renames) build system code, tools and documentation related to `prodtype`.

#### Rationale:

- We don't need the code that has no effect on the build process.

#### Review Hints:

- There are notes in the PR I left to discuss or explain the changes.
- Despite the fact that platforms (OVAL and remediation metadata) are somewhat related to prodtype, their improvement is considered out of scope for this PR.
- Automatus tests are failing because they can't satisfy scenario requirements for packages. Which should not be a surprise as an attempt to test RHEL7 content in Fedora is pretty much doomed to fail.
- **Pay attention to 3-rd commit**: https://github.com/ComplianceAsCode/content/pull/11505/commits/a623596c6741209bd10d31d77403a4342144d173. It could start doing things we don't actually want.

Closes: #11382.

### CODE REVIEW

File: ssg/fixes.py [hunk 1]
```diff
@@ -48,10 +47,10 @@ def applicable_platforms(fix_path):
     if 'platform' not in config:
         raise ValueError("Malformed fix: missing platform" % fix_path)
 
-    return parse_prodtype(config['platform'])
+    return parse_platform(config['platform'])
 
 
-def parse_platform(fix_contents):
+def find_platform_line(fix_contents):
```
Comment (evgenyz) [inline ssg/fixes.py:53]: Better reflects what it actually does.

File: ssg/utils.py [hunk 2]
```diff
@@ -179,9 +179,17 @@ def parse_name(product):
...
 
 
+def parse_platform(platform):
+    """
+    From a platform line, returns the set of platforms listed.
+    """
+
+    return set(map(lambda x: x.strip(), platform.split(',')))
+
+
```
Comment (evgenyz) [inline ssg/utils.py:189]: It has been transferred from the `build_yaml` module and renamed to reflect the scope of this new-old function, that is now used only to parse platforms.

File: tests/ssg_test_suite/common.py [hunk 3]
```diff
@@ -309,19 +308,13 @@ def load_rule_and_env(rule_dir_path, env_yaml, product=None):
...
-    prodtypes = parse_prodtype(rule.prodtype)
-
     # Our local copy of env_yaml needs some properties from rule.yml
     # for completeness.
     local_env_yaml = dict()
     local_env_yaml.update(env_yaml)
     local_env_yaml['rule_id'] = rule.id_
     local_env_yaml['rule_title'] = rule.title
-    local_env_yaml['products'] = prodtypes
+    local_env_yaml["products"] = {product}
```
Comment (Mab879) [inline tests/ssg_test_suite/common.py:317]:
```suggestion
    local_env_yaml['products'] = {product}
```

**Review by Mab879 — APPROVED** [review]
Overall, looks good.

Just more minor change that I'm not willing to block over.

File: linux_os/guide/services/ftp/ftp_configure_vsftpd/ftp_configure_firewall/rule.yml [hunk 4]
```diff
@@ -21,7 +21,7 @@ description: |-
 rationale: |-
     These settings configure the firewall to allow connections to an FTP server.
 
-    {{% if prodtype != "rhel7" %}}
+    {{% if product != "rhel7" %}}
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply linux_os/guide/services/ftp/ftp_configure_vsftpd/ftp_configure_firewall/rule.yml:24]: \o hello @evgenyz -- Good question! It has been so long, I do not know. Likely I made a mistake and expected the build system to catch it...

**Comment by evgenyz** [comment]
> Overall, looks good.
> 
> Just more minor change that I'm not willing to block over.

Could you please check the 3-rd commit and maybe provide some insight about the change: is it going to do a good thing or a bad thing?

**Comment by Mab879** [comment]
> > Overall, looks good.
> > Just more minor change that I'm not willing to block over.
> 
> Could you please check the 3-rd commit and maybe provide some insight about the change: is it going to do a good thing or a bad thing?

They don't look actively harmful.

File: docs/manual/developer/06_contributing_with_content.md [hunk 5]
```diff
@@ -518,9 +518,9 @@ the use of unversioned products here (e.g., `rhel` applies to `rhel7`,
 correct extension for content of that type (e.g., `.sh` for `bash`
 content). Further, all of these directories are optional and will only
 be searched for content if present. Lastly, the product naming of
-content will not override the contents of `platform` or `prodtype`
-fields in the content itself (e.g., if `rhel7` is not present in the
-`rhel7.xml` OVAL check platform specifier, it will be included in the
+content will not override the contents of `platform` field in
+the content itself (e.g., if `rhel7` is not present in the `rhel7.xml`
```
[the remark this answers is not in the document]
Comment (evgenyz) [reply docs/manual/developer/06_contributing_with_content.md:522]: Yep. I cleaned up connections between prodtype and platforms, but kept them out of the scope. It is a zoo there.

File: docs/manual/developer/03_creating_content.md [hunk 6]
```diff
@@ -742,9 +736,6 @@ between XCCDF rules which directly implement the given controls (represented by
 The `rules` and `related_rules` keys consist of a list of rule IDs and variable
 selections.
 
-If a rule needs to be chosen only in some of products despite its `prodtype` we
-can use Jinja macros inside the controls file to choose products.
```
[the remark this answers is not in the document]
Comment (evgenyz) [reply docs/manual/developer/03_creating_content.md:746]: Fixed.

**Review by jan-cerny — APPROVED** [review]
I have reviewed code.  I also run Automatus locally.

**Pull request merged**